### PR TITLE
Clean up build pipelines

### PR DESCRIPTION
**Pull request by calebmshafer**
- Fix Node 12 compatibility issues with using `fs.rmSync`. This API did not get added until v14.14 and v16.14.
- Fix the public ci build
- Ensure consistent Node versions
  - The regression testing was checking the latest Node version (18) instead of that being the main CI build.
- Redundant info being set in the fast-ci.yaml vs. ci.yaml
- Update minimum Node 12 version tested with, only support >12.22 so no need to test earlier versions.
- Allow integration regression to scale out to as many agents as possible for quick turnaround. Uses standard Azure pool, not the custom pool.

The following builds are not run directly on PR builds but do need to be validated as they're performing regression testing and other validation that takes longer than we'd like to wait for each one.

- Regression build
- Integration tests
- Integration regression tests
- iOS tests

Regression test run: https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=1926713&view=results
Integration regression: https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=1926767&view=results

### CODE REVIEW

**Comment by calebmshafer** [comment]
@raplemie @GerardasB @NancyMcCallB looks like there are a couple issues with the CI now that I switched it to the latest Node 18 version. Can you take a look?

```
--[ FAILURE: @itwin/components-react ]--------------[ 1 minute 20.2 seconds ]--

undefined
Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: ReactDataGrid, Viewport, t
Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

  ...285 lines omitted...
      at waitForWrapper (D:\vsts_a\4\s\common\temp\node_modules\.pnpm\@testing-library+dom@8.20.0\node_modules\@testing-library\dom\dist\wait-for.js:166:27)
      at Context.<anonymous> (src\test\editors\DateTimeEditor.test.tsx:257:25)
      at processImmediate (node:internal/timers:471:21)



ERROR: Coverage for lines (99.89%) does not meet global threshold (100%)
ERROR: Coverage for functions (99.89%) does not meet global threshold (100%)
ERROR: Coverage for branches (99.93%) does not meet global threshold (100%)
ERROR: Coverage for statements (99.9%) does not meet global threshold (100%)

--[ FAILURE: @itwin/imodel-components-react ]--------[ 1 minute 7.2 seconds ]--


  288 passing (30s)
  1 failing

  1) <TimelineComponent showDuration={true} />
       should respect time zone offset:

      AssertionError: expected '07:00:00\u202fPM' to equal '07:00:00 PM'
      + expected - actual

      -07:00:00 PM
      +07:00:00 PM
      
      at Context.<anonymous> (src\test\timeline\TimelineComponent.test.tsx:889:41)
      at processImmediate (node:internal/timers:471:21)
```

**Review by grigasp — APPROVED** [review]

File: common/config/azure-pipelines/jobs/regression-testing.yaml [hunk 1]
```diff
@@ -110,6 +110,7 @@ jobs:
       maxParallel: 3
```
Comment (aruniverse) [inline common/config/azure-pipelines/jobs/regression-testing.yaml:110]: do we really need to run for node 12.22 and another for 12.x?

Comment (calebmshafer) [reply common/config/azure-pipelines/jobs/regression-testing.yaml:94]: fixed.

Comment (aruniverse) [reply common/config/azure-pipelines/jobs/regression-testing.yaml:94]: sorry, another thing, does it make sense to run regression for node18, if thats the main ver we use in our ci? 
imo wastes another 30 mins or so, which maybe is neglibile

Comment (calebmshafer) [reply common/config/azure-pipelines/jobs/regression-testing.yaml:94]: I still think it's important to test both the first LTS version of a major (18.12) and the latest one.  In this PR I switched around our regular ci build to use the latest, 18.x, and the regression tests to check the first Major.minor that we support.

Comment (aruniverse) [reply common/config/azure-pipelines/jobs/regression-testing.yaml:94]: > We have run into compatibility issues previously due to this.

issues in the same minor? thats very concerning...

Comment (aruniverse) [reply common/config/azure-pipelines/jobs/regression-testing.yaml:94]: i guess the appui issue from above is an example. resolving this comment

**Review by DanRod1999 — APPROVED** [review]

**Comment by aruniverse** [comment]
> @raplemie @GerardasB @NancyMcCallB looks like there are a couple issues with the CI now that I switched it to the latest Node 18 version. Can you take a look?
> 
> ```
> --[ FAILURE: @itwin/components-react ]--------------[ 1 minute 20.2 seconds ]--
> 
> undefined
> Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.
> 
> * Move data fetching code or side effects to componentDidUpdate.
> * If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state
> * Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
> 
> Please update the following components: ReactDataGrid, Viewport, t
> Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.
> 
>   ...285 lines omitted...
>       at waitForWrapper (D:\vsts_a\4\s\common\temp\node_modules\.pnpm\@testing-library+dom@8.20.0\node_modules\@testing-library\dom\dist\wait-for.js:166:27)
>       at Context.<anonymous> (src\test\editors\DateTimeEditor.test.tsx:257:25)
>       at processImmediate (node:internal/timers:471:21)
> 
> 
> 
> ERROR: Coverage for lines (99.89%) does not meet global threshold (100%)
> ERROR: Coverage for functions (99.89%) does not meet global threshold (100%)
> ERROR: Coverage for branches (99.93%) does not meet global threshold (100%)
> ERROR: Coverage for statements (99.9%) does not meet global threshold (100%)
> 
> --[ FAILURE: @itwin/imodel-components-react ]--------[ 1 minute 7.2 seconds ]--
> 
> 
>   288 passing (30s)
>   1 failing
> 
>   1) <TimelineComponent showDuration={true} />
>        should respect time zone offset:
> 
>       AssertionError: expected '07:00:00\u202fPM' to equal '07:00:00 PM'
>       + expected - actual
> 
>       -07:00:00 PM
>       +07:00:00 PM
>       
>       at Context.<anonymous> (src\test\timeline\TimelineComponent.test.tsx:889:41)
>       at processImmediate (node:internal/timers:471:21)
> ```

https://github.com/nodejs/node/issues/46123

**Review by aruniverse — APPROVED** [review]

**Pull request merged**